src/App.js

```diff
@@ -1,26 +1,144 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import React from "react";
+import "./App.css";
+import TodoItems from "./todoitems";
+import Filters from "./filters";
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      textLines: {
+        hello: false,
+        bye: false,
+        "another task": false,
+        "last task": false
+      },
+      completedItems: 0,
+      filter: null,
+      displayStyle: { display: "block" },
+      isSelect: ["all", "active", "completed"]
+    };
+  }
+
+  completingItems = () => {
+    this.setState(prevState => {
+      return { completedItems: prevState.completedItems + 1 };
+    });
+  };
+
+  leftIems = () => {
+    return Object.keys(this.state.textLines).length - this.state.completedItems;
+  };
+
+  addTask = event => {
+    if (event.key === "Enter") {
+      let task = event.target.value;
+      event.target.value = "";
+      this.setState(prevState => {
+        const tmp = { ...prevState.textLines };
+        tmp[task] = false;
+        return { textLines: tmp, displayStyle: { display: "block" } };
+      });
+    }
+  };
+
+  changeDisplay = isView => {
+    const newDisplay = isView ? { display: "none" } : { display: "block" };
+    this.setState({ displayStyle: newDisplay });
+  };
+
+  removeTask = key => {
+    this.setState(prevState => {
+      const tmp = { ...prevState.textLines };
+      delete tmp[key];
+      return { textLines: tmp };
+    });
+  };
+
+  isComplete = key => {
+    this.setState(prevState => {
+      const tmp = { ...prevState.textLines };
+      tmp[key] = !prevState.textLines[key];
+      return { textLines: tmp };
+    });
+  };
+
+  setFilter = newFilter => {
+    this.setState({
+      filter: newFilter
+    });
+  };
+
+  clearComplete = () => {
+    this.setState(prevState => {
+      const tmp = { ...prevState.textLines };
+      for (let key in tmp) {
+        if (tmp[key]) {
+          delete tmp[key];
+        }
+      }
+      return { textLines: tmp, completedItems: 0 };
+    });
+  };
+
+  render() {
+    if (this.state.displayStyle.display === "block") {
+      if (Object.keys(this.state.textLines).length === 0) {
+        this.setState({ displayStyle: { display: "none" } });
```

Comment:
I'd recommend avoid calling `setState` directly in render. It should be called when some event was fired. But not on each `render`. Even if it is encapsulated in `if => if => setState()` chain